File: Python/largest-number.py
class Solution:
    # @param num, a list of integers
    # @return a string
    def largestNumber(self, num):
        def cmpstr(a, b):
            return -1 if a+b>b+a else 1
            
        num = sorted([str(x) for x in num], cmp = cmpstr)
        
        ans = ''.join(num).lstrip('0')
            
        return ans or '0'

File: C++/largest-number.cpp
class Solution {
private:
    static bool cmp(int a, int b) {
        std::string m=to_string(a), n=to_string(b);
        return m+n>n+m;
    }
public:
    string largestNumber(std::vector<int>& nums) {
        std::string ans;
        std::sort(nums.begin(), nums.end(), cmp);
        for(int n : nums) ans+=std::to_string(n);
        return ans.front()!='0' ? ans : "0";
    }
};

